### ISSUE TITLE
PETSc SolverControl: Maximum number of iterations truncated?

### ISSUE THREAD

**Issue by marcfehling** [dealii/dealii#8733]
As noticed in #8729, `PETSc` does not seem to follow the user's specification on the maximum number of iterations given by the `SolverControl` object: It forces an own maximum of `10000` iterations if the user's value exceeds this threshold. The question came up whether `PETSc` sticks to some own default value, or truncates it to some own maximum value.

We should either overwrite these defaults (if they exist), or warn the user that their provided maximum number of iterations will be overwritten by `PETSc`.

**Comment by tjhei** [comment]
I don't think it is useful to do more than 10000 iterations, but 10000 is the default for PETSc. This should be fixable by calling KSPSetTolerances with a very large number for the last argument (similar to what we do for SolverRichardson)
